pysat/_instrument.py

```diff
@@ -1065,18 +1065,17 @@ def __setitem__(self, key, new_data):
                         # 'Epoch'
                         self.data[key] = (epoch_name, in_data)
                     elif len(in_data) == 1:
-                        # Only provided a single number in iterable, make that
-                        # the input for all times
-                        self.data[key] = (epoch_name,
-                                          [in_data[0]] * len(self.index))
+                        # Only provided a single number in iterable.
+                        # Assign as coord.
+                        self.data.coords[key] = (key, [in_data[0]])
                     elif len(in_data) == 0:
                         # Provided an empty iterable, make everything NaN
                         self.data[key] = (epoch_name,
                                           [np.nan] * len(self.index))
                 elif len(np.shape(in_data)) == 0:
-                    # Not an iterable input, rather a single number.  Make
-                    # that number the input for all times.
-                    self.data[key] = (epoch_name, [in_data] * len(self.index))
+                    # Not an iterable input, but a single number.
+                    # Assign as coord.
```

Comment:
```suggestion
                    # Not an iterable input, but a single number.
                    # To avoid padding over time, assign as a coord.
```